tests/components/androidtv/test_remote.py

```diff
@@ -0,0 +1,147 @@
+    return_value=None,
+) -> None:
+    """Test generic Android media player entity service."""
+    service_data = {ATTR_ENTITY_ID: entity_id}
+    if additional_service_data:
+        service_data.update(additional_service_data)
+
+    androidtv_patch = (
+        "androidtv.androidtv_async.AndroidTVAsync"
+        if "android" in entity_id
+        else "firetv.firetv_async.FireTVAsync"
+    )
+    with patch(
+        f"androidtv.{androidtv_patch}.{androidtv_method}", return_value=return_value
+    ) as service_call:
+        await hass.services.async_call(
+            REMOTE_DOMAIN,
+            ha_service_name,
+            service_data=service_data,
+            blocking=True,
+        )
+        assert service_call.called
+
+
+@pytest.mark.parametrize("config", [CONFIG_ANDROID_DEFAULT, CONFIG_FIRETV_DEFAULT])
+async def test_services_remote(hass: HomeAssistant, config) -> None:
+    """Test services for remote entity."""
+    patch_key, entity_id, config_entry = _setup(config)
+    config_entry.add_to_hass(hass)
+
+    with patchers.patch_connect(True)[patch_key]:
+        with patchers.patch_shell(SHELL_RESPONSE_OFF)[patch_key]:
+            assert await hass.config_entries.async_setup(config_entry.entry_id)
+            await hass.async_block_till_done()
+
+        with (
+            patchers.patch_shell(SHELL_RESPONSE_STANDBY)[patch_key],
+            patchers.PATCH_SCREENCAP,
+        ):
+            await _test_service(hass, entity_id, SERVICE_TURN_OFF, "turn_off")
+            await _test_service(hass, entity_id, SERVICE_TURN_ON, "turn_on")
+            await _test_service(
+                hass,
+                entity_id,
+                SERVICE_SEND_COMMAND,
+                "adb_shell",
+                {ATTR_COMMAND: ["BACK", "test"], ATTR_NUM_REPEATS: 2},
+            )
+
+
+@pytest.mark.parametrize("config", [CONFIG_ANDROID_DEFAULT, CONFIG_FIRETV_DEFAULT])
+async def test_services_remote_custom(hass: HomeAssistant, config) -> None:
+    """Test services with custom options for remote entity."""
+    patch_key, entity_id, config_entry = _setup(config)
+    config_entry.add_to_hass(hass)
+    hass.config_entries.async_update_entry(
+        config_entry,
+        options={
+            CONF_TURN_OFF_COMMAND: "test off",
+            CONF_TURN_ON_COMMAND: "test on",
+        },
+    )
+
+    with patchers.patch_connect(True)[patch_key]:
+        with patchers.patch_shell(SHELL_RESPONSE_OFF)[patch_key]:
+            assert await hass.config_entries.async_setup(config_entry.entry_id)
+            await hass.async_block_till_done()
+
+        with (
+            patchers.patch_shell(SHELL_RESPONSE_STANDBY)[patch_key],
+            patchers.PATCH_SCREENCAP,
+        ):
+            await _test_service(hass, entity_id, SERVICE_TURN_OFF, "adb_shell")
+            await _test_service(hass, entity_id, SERVICE_TURN_ON, "adb_shell")
+
+
+async def test_remote_unicode_decode_error(hass: HomeAssistant) -> None:
```

Comment:
It's not clear to me how the tests shows how the `UnicodeDecodeError` is handled.

Reply:
This was because the decorator normally set the entity as not available, but for sure was missing a `cap_log` check. Anyway I changed the logic to raise a `ServiceValidationError` and so I changed the tests